plugins/channel_header_plug/index.js

```diff
@@ -6,9 +6,13 @@ import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
 import ChannelHeaderPlug from './channel_header_plug.jsx';
 
-function mapStateToProps(state) {
+function mapStateToProps(state, ownProps) {
+    let components = state.plugins.components.ChannelHeaderButton;
+    if (ownProps && ownProps.callPlugins) {
+        components = state.plugins.components.ChannelHeaderCallButton;
```

Comment:
Not something to do now, but if the call button version of this ever starts to require more complicated to get its props, we could also consider splitting this up so that the call button has its own `mapStateToProps`, but it still shares the same underlying `ChannelHeaderPlug` component. Using a prop to switch between the two components is good enough for now though.